packages/rest/src/validation/request-body.validator.ts

```diff
@@ -79,11 +56,27 @@ function convertToJsonSchema(openapiSchema: SchemaObject) {
   return jsonSchema;
 }
 
-// tslint:disable-next-line:no-any
-function validateValueAgainstJsonSchema(body: any, schema: any) {
-  const ajv = new AJV({allErrors: true});
+function validateValueAgainstJsonSchema(
+  // tslint:disable-next-line:no-any
+  body: any,
+  // tslint:disable-next-line:no-any
+  schema: any,
+  globalSchemas?: SchemasObject,
+) {
+  // from @miroslav: A temporary solution for resolving schema references produced
+  // by @loopback/repository-json-schema. In the future, we should
+  // support arbitrary references anywhere in the OpenAPI spec.
+  // See https://github.com/strongloop/loopback-next/issues/435
+  let schemaWithComponents = _.cloneDeep(schema);
+  schemaWithComponents.components = {
+    schemas: globalSchemas,
```

Comment:
Should we set a sensible default for `schemas` here if `globalSchemas` is undefined (since it's an optional parameter)?

Reply:
@b-admike I think it's fine to have `schemas` as `undefined` :) It's still a valid json schema, you can treat `components` as an arbitrary additional property, added only for reference's usage.

On a second thought, I feel `schemaWithRef` would be a better name for it. Renamed in the new commit.